docs/ci-learnings.md

```
@@ -665,3 +665,20 @@ if s["fail"] == 0 and s["warn"] == 0:
 **PR/Commit:** N/A
+
+
+---
+
+## [2026-05-16] ci — Dependency-Check stale H2 cache lock
+
+**Symptom:** OWASP Dependency-Check nightly failed on current main before producing reports with 'Unable to obtain an exclusive lock on the H2 database' and 'No documents exist'.
+
+**Root cause:** The restored ~/.dependency-check/data cache can contain stale H2 lock/trace files from a prior run, preventing the scanner container from updating the local NVD database.
+
+**Fix applied:** Clean stale *.lock.db and *.trace.db files immediately after restoring the Dependency-Check cache and before running the scanner; classify the H2 lock message as infrastructure-flavoured for retry handling; bound the scanner command with a 30-minute timeout and hard kill so verification does not require waiting through a full nightly window.
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside emphasis markers

(MD037, no-space-in-emphasis)